homeassistant/components/feedreader.py

```diff
@@ -60,22 +61,46 @@ def _update(self):
             elif len(self._feed.entries) > 0:
                 _LOGGER.debug('Entries available in feed %s', self._url)
                 self._publish_new_entries()
-                self._last_entry_timestamp = \
-                    self._feed.entries[0].published_parsed
             else:
                 self._log_no_entries()
+        _LOGGER.info('Fetch from feed "%s" completed', self._url)
+
+    def _update_and_fire_entry(self, entry):
+        """Update last_entry_timestamp and fire entry."""
+        # We are lucky, `published_parsed` data available,
+        # let's make use of it to publish only new available
+        # entries since the last run
+        if 'published_parsed' in entry:
```

Comment:
Ok you're right, it's not a simple dict but some kind of subset of it, `feedparser.FeedParserDict`
I've tested this before submitting with different feeds, i.e. one with `published_parsed` support and one not, and works fine.
Not sure how to write proper tests here, apparently depending on the kind of feed some parameters are available and others are not.
How do you suggest to proceed?